warehouse/index.js

```diff
@@ -92,6 +95,40 @@ function excludeRudderCreatedTableNames(
   return name;
 }
 
+function appendColumnNameAndType(
+  utils,
+  eventType,
+  key,
+  val,
+  output,
+  columnTypes,
+  options,
+  jsonKey = false
+) {
+  const datatype = getDataType(key, val, options, jsonKey);
+  val = jsonKey && isJson(val) ? JSON.stringify(val) : val;
```

Comment:
The JSON check is already made in the above statement.
```suggestion
  val = datatype === "json" ? JSON.stringify(val) : val;
```

Reply:
Can the value of `datatype` be `json` even if `val` is not a JSON?

Reply:
No, at the time loading data into warehouse like snowflake, it will throw error if correct json string is not passed and column declared as json(variant in DB)

Reply:
I'm asking if we can rely on the value of the `datatype` variable so that the original check I proposed can be used. Otherwise, we can skip this.

Reply:
logic has been changed. Only one time validation is done